server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -258,6 +240,23 @@
         Setting.longSetting("index.soft_deletes.retention.operations", 0, 0,
             Property.IndexScope, Property.Dynamic);
 
+    /**
+     * Controls how long translog files that are no longer needed for persistence reasons
+     * will be kept around before being deleted. This setting will be ignored if soft-deletes is enabled.
+     **/
+    public static final Setting<TimeValue> INDEX_TRANSLOG_RETENTION_AGE_SETTING =
+        Setting.timeSetting("index.translog.retention.age",
+            settings -> INDEX_SOFT_DELETES_SETTING.get(settings) ? TimeValue.MINUS_ONE : TimeValue.timeValueHours(12), TimeValue.MINUS_ONE,
+            Property.Dynamic, Property.IndexScope);
+
+    /**
+     * Controls how many translog files that are no longer needed for persistence reasons
+     * will be kept around before being deleted. This setting will be ignored if soft-deletes is enabled.
```

Comment:
you've deleted the last sentence in the prior Javadocs (i.e.  `Keeping more files is useful to increase the chance of ops based recoveries`). Perhaps keep that and qualify it that it only applies to non-soft-deleted indices. Same goes for the other settting